### PR TITLE
Fix `SETTINGS_HEADER_TABLE_SIZE` implementation

### PR DESCRIPTION
**Pull request by rschmitt**

### CODE REVIEW

**Comment by michael-o** [comment]
That one looks obvious. Looks like copy and paste.

File: httpcore5-h2/src/main/java/org/apache/hc/core5/http2/config/H2Config.java [hunk 1]
```diff
@@ -121,8 +121,7 @@ public String toString() {
                 .setPushEnabled(INIT_ENABLE_PUSH)
                 .setMaxConcurrentStreams(Integer.MAX_VALUE) // no limit
                 .setMaxFrameSize(INIT_MAX_FRAME_SIZE)
-                .setInitialWindowSize(INIT_WINDOW_SIZE)
-                .setHeaderTableSize(Integer.MAX_VALUE); // unlimited
+                .setInitialWindowSize(INIT_WINDOW_SIZE);
```
Comment (ok2c) [inline httpcore5-h2/src/main/java/org/apache/hc/core5/http2/config/H2Config.java]: @rschmitt `.setHeaderTableSize(Integer.MAX_VALUE); // unlimited` should be `.setMaxHeaderListSize(Integer.MAX_VALUE); // unlimited`

Mea culpa

File: httpcore5-h2/src/main/java/org/apache/hc/core5/http2/impl/nio/AbstractH2StreamMultiplexer.java [hunk 2]
```diff
@@ -1188,7 +1188,7 @@ private void produceOutput() throws HttpException, IOException {
     private void applyRemoteSettings(final H2Config config) throws H2ConnectionException {
         remoteConfig = config;
 
-        hPackDecoder.setMaxTableSize(remoteConfig.getMaxHeaderListSize());
+        hPackEncoder.setMaxTableSize(remoteConfig.getHeaderTableSize());
```
Comment (ok2c) [inline httpcore5-h2/src/main/java/org/apache/hc/core5/http2/impl/nio/AbstractH2StreamMultiplexer.java]: @rschmitt Mea cupla

File: httpcore5-h2/src/main/java/org/apache/hc/core5/http2/config/H2Config.java (hunk 1 above)
Comment (rschmitt) [reply httpcore5-h2/src/main/java/org/apache/hc/core5/http2/config/H2Config.java]: Fair enough, but this option is not actually implemented.

Comment (ok2c) [reply httpcore5-h2/src/main/java/org/apache/hc/core5/http2/config/H2Config.java]: > Fair enough, but this option is not actually implemented.

I am presently working on it unless you want me to focus on something more important blocking the GA release.

Comment (rschmitt) [reply httpcore5-h2/src/main/java/org/apache/hc/core5/http2/config/H2Config.java]: I don't have a root cause or even a detailed problem description yet, but I've observed that `SubscribeToShardIntegrationTest` fails when `H2Config#setInitialWindowSize` is called with a non-default window size. In fact, even setting the initial window size to 65,534 (i.e. one less than the default) seems to consistently result in `FLOW_CONTROL_ERROR`s from the server. This is the most severe bug I'm currently able to reproduce. (The `IOSession` shutdowns are even more concerning, but I don't have a repro case at the moment.)

**Pull request merged**